prog.go

```diff
@@ -419,7 +419,7 @@ func unmarshalProgram(buf []byte) (*Program, error) {
 	// Looking up an entry in a nested map or prog array returns an id,
 	// not an fd.
 	id := internal.NativeEndian.Uint32(buf)
-	fd, err := bpfGetProgramFDByID(id)
+	fd, err := objGetFDByID(_ProgGetFDByID, id)
 	if err != nil {
```

Comment:
You can replace the rest of this function with a call to NewProgramFromFD, same for unmarshalMap.

Reply:
updated this functions as well.